cache/class-vary-cache.php

```diff
@@ -211,14 +212,17 @@ private static function decrypt_cookie_value( $cookie_value ) {
 	 * @access  private
 	 */
 	private static function parse_group_cookie() {
-		if ( isset( $_COOKIE[ self::COOKIE_SEGMENT ] ) ) {
-			$cookie_value = sanitize_text_field( wp_unslash( $_COOKIE[ self::COOKIE_SEGMENT ] ) );
+		if ( isset( $_COOKIE[ self::COOKIE_SEGMENT ] ) || isset( $_COOKIE[ self::COOKIE_AUTH ] ) ) {
```

Comment:
Edge case but we could run into potential complications here where a user has an auth cookie but encryption not enabled (also vice versa). It would be better to more explicitly match cases:

```
if ( self::is_encryption_enabled() && ! empty( $_COOKIE[ self::COOKIE_AUTH ] ) ) {
    $cookie_value = self::decrypt_cookie_value( $_COOKIE[ self::COOKIE_AUTH ] );
} elseif ( ! empty( $_COOKIE[ self::COOKIE_SEGMENT ] ) ) {
    $cookie_value = $_COOKIE[ self::COOKIE_SEGMENT ];
}

if ( empty( $cookie_value ) {
    return;
}

// strip version
// parse
```

Reply:
This occurred to me when I was writing the original logic but this is much simpler check than what I had in mind.